src/rest/methods/notification.ts

```diff
@@ -0,0 +1,145 @@
+import { dateToString, stringToDate } from '../../utils/stringToDate'
+import { InterfaceAllthingsRestClient } from '../types'
+
+export enum EnumNotificationCategory {
+  events = 'events',
+  hintsAndTips = 'hints-and-tips',
+  lostAndFound = 'lost-and-found',
+  localDeals = 'local-deals',
+  localEvents = 'local-events',
+  miscellaneous = 'miscellaneous',
+  deals = 'deals',
+  messages = 'messages',
+  adminMessages = 'admin-messages',
+  newThingsToGive = 'new-things-to-give',
+  newThingsForSale = 'new-things-for-sale',
+  surveys = 'surveys',
+  supportOffer = 'support-offer',
+  supportRequest = 'support-request',
+  sustainability = 'sustainability',
+  localServices = 'local-services',
+  services = 'services',
+  ticketDigestEmail = 'ticket-digest-email',
+  appDigestEmail = 'app-digest-email',
+  newFile = 'new-file',
+}
+
+export enum EnumNotificationType {
+  clipboardThing = 'clipboard-thing',
+  comment = 'comment',
+  communityArticle = 'community-article',
+  newFile = 'new-file',
+  ticketComment = 'ticket-comment',
+  welcomeNotification = 'welcome-notification',
+}
+
+export interface IBasicNotification {
+  readonly category: EnumNotificationCategory
+  readonly id: string
+  readonly read: boolean
+  readonly title: string
+  readonly type: EnumNotificationType
+}
+
+export interface IPreprocessedNotification extends IBasicNotification {
+  readonly createdAt: string
+  readonly objectID: string
+  readonly referencedObjectID: string | null
+}
+
+export interface INotification extends IBasicNotification {
+  readonly createdAt: Date
+  readonly objectId: string
+  readonly referencedObjectId: string | null
+}
+
+const remapNotificationResult = (
+  notification: IPreprocessedNotification,
+): INotification => {
+  const {
+    category,
+    id,
+    read,
+    title,
+    type,
+    createdAt,
+    objectID,
+    referencedObjectID,
+  } = notification
+
+  return {
```

Comment:
Since you are not remapping all the objects in the notifications result, you can just destructure the object keys you need and use the ...rest syntax to reduce the excess line numbers.

ie: 
```suggestion
    return {
    createdAt: stringToDate(createdAt),
    objectId: objectID,
    referencedObjectId: referencedObjectID,
    ...rest
    }
```